core-processor/src/ext.rs

```diff
@@ -737,7 +737,16 @@ impl Externalities for Ext {
     }
 
     fn debug(&self, data: &str) -> Result<(), Self::UnrecoverableError> {
-        log::debug!(target: "gwasm", "DEBUG: {}", data);
+        let program_id = self.program_id()?;
+        let dispatch_kind = self.context.message_context.dispatch_kind();
+        let entry_point = dispatch_kind.as_entry();
+        let message_id = self.message_id()?;
+
+        log::trace!(target: "gwasm", "Program {program_id} executes {entry_point:?}");
+        log::trace!(target: "gwasm", "Current message id: {message_id}");
+
+        log::debug!(target: "gwasm", "DEBUG: {data}");
```

Comment:
I was considering something like this in logs as gwasm target:
`[handle (0xMS..ID)] 0xPR..ID: {data}`